File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/event/function/FunctionPreParseEvent.java
package me.lokka30.levelledmobs.bukkit.event.function;

import me.lokka30.levelledmobs.bukkit.logic.LmFunction;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public final class FunctionPreParseEvent extends CancellableFunctionEvent {

    /* vars */

    private static final HandlerList HANDLERS = new HandlerList();

    /* constructors */

    public FunctionPreParseEvent(final @NotNull LmFunction function) { super(function); }

    /* getters and setters */

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/function/FunctionPreParseEvent.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.function

import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class FunctionPreParseEvent(
    val function: LmFunction
) : Event() {
    companion object{
        val HANDLERS = HandlerList()
    }
    var cancelled = false

    override fun getHandlers(): HandlerList{
        return HANDLERS
    }
}